habitat_sim/utils/profiling_utils.py

```diff
@@ -44,21 +68,96 @@ def my_function():
     from torch.cuda import nvtx
 
 
+@attr.s(auto_attribs=True)
+class _ProfilingHelper:
+    step_count: int = -1
+    capture_start_step: int = -1
+    capture_end_step: int = -1
+    range_depth: int = 0
+
+
+_helper = _ProfilingHelper()
+
+
+def configure(capture_start_step=-1, num_steps_to_capture=-1):
+    r"""Configures profiling_utils to push a "habitat_capture_range" range to
+    span the desired train steps. See also on_start_step(). See
+    profiling_utils.py file-level documentation for example usage with Nvidia
+    Nsight.
+
+    Your program can have many processes that call range_push/range_pop, but
+    only your main process (the one executing your train loop) should call
+    configure() and on_start_step().
+
+    If capture_start_step == 0, then capturing starts at the start of step 0
+    (after initialization code). If capture_start_step == -1, then capturing
+    starts immediately after your call to configure() and includes any
+    initialization code that runs prior to step 0. If num_steps_to_capture ==
+    -1, the range won't be popped and capture will continue until your program
+    terminates.
+    """
+
+    if not enable_profiling:
```

Comment:
Is "enable_profiling"  a constant restricted to this module?  If so, would it not be better to have it named 
`_ENABLE_PROFILING` 

as per PEP? (with single leading underscore acting sort of like protected access mod - not sure if this variable is to be imported when this module is imported)

Reply:
I wouldn't consider this a constant based on my usage: the value is determined at runtime and varies from run to run. But I agree about it being "internal use" so I'll add the underscore.